cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +131,84 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    fieldValidator.validateField(orderByField);
+    if (!schema.isIndexColumn(orderByField.getName())) {
+      throw new InvalidFieldException(schema.getTableId(), orderByField.getName(), "is not an indexed column");
+    }
+    ScannerIterator scannerIterator = new ScannerIterator(getScanner(keyRange), schema);
+    LimitIterator limitIterator = new LimitIterator(Collections.singleton(scannerIterator).iterator(),
```

Comment:
can't limit it here, otherwise it can return wrong values if there are more than limit rows. Should be wrapping the PriorityQueue's iterator.

Reply:
also be sure to add a unit test for this scenario

Reply:
nice catch!